core/api/src/main/java/org/trellisldp/api/MutableDataService.java

```diff
@@ -27,27 +28,39 @@
 public interface MutableDataService<U> extends RetrievalService<U> {
 
     /**
-     * @param identifier the identifier for the resource to persist
+     * Put a resource into the server.
+     *
+     * @param identifier the identifier for the new resource
      * @param session the session context for this operation
-     * @param resource a resource to persist
-     * @return whether the resource was successfully persisted
+     * @param ixnModel the LDP interaction model for this resource
+     * @param container an LDP container for this resource
+     * @param dataset the dataset
+     * @return whether the resource was added
      */
-    Future<Boolean> create(IRI identifier, Session session, U resource);
+    Future<Boolean> create(IRI identifier, Session session, IRI ixnModel, IRI container, Dataset dataset);
```

Comment:
See above about removing the generic `<U>` type from the interface definition, since it appears that it is no longer used.